client/tests/gui/source/delete/test_dialog.py

```diff
@@ -1,22 +1,91 @@
+
+class TestDeleteSourceDialog:
+    def test_dialog_setup(self, dialog):
+        assert type(dialog) is DeleteSourceDialog
+        assert type(dialog.sources) is set
+        assert dialog.dangerous
+
+    def test_default_button_is_safer_choice(self, dialog):
         # This test does rely on an implementation detail (the buttons)
         # but I couldn't find a way to test this properly using key events.
-        assert not self.dialog.continue_button.isDefault()
-        assert self.dialog.cancel_button.isDefault()
+        assert not dialog.continue_button.isDefault()
+        assert dialog.cancel_button.isDefault()
+
+    def test_displays_important_information_when_shown(self, dialog):
+        if len(dialog.sources) < 1:
+            pytest.skip("Skip if no sources")
+        assert "not be able to send them replies" in dialog.text()
+        assert "source will not be able to log in" in dialog.text()
+        assert "files and messages from that source will also be destroyed" in dialog.text()
+
+    def test_dialog_continue_button_adapts_to_source_count(self, dialog):
+        count = len(dialog.sources)
+
+        if count < 1:
+            pytest.skip("Skip if no sources")
+
+        if count == 1:
+            assert dialog.continue_button.text() == "YES, DELETE ENTIRE SOURCE ACCOUNT"
+        elif count > 1:
+            assert dialog.continue_button.text() == f"YES, DELETE {count} SOURCE ACCOUNTS"
+        else:
+            # should be unreachable due to decorator
+            pytest.fail("Unreachable, or so we thought")
+
+    def test_no_sources_shows_error_text(self, dialog):
+        if len(dialog.sources) > 0:
+            pytest.skip("Skip if sources")
+
+        assert dialog.text() == "No sources have been selected."
+
+    def test_no_sources_continue_button_disabled(self, dialog):
+        if len(dialog.sources) > 0:
+            pytest.skip("Skip if sources")
+
+        assert not dialog.continue_button.isEnabled()
+
+    def test_correct_format_body_text(self):
+        """
+        For n > 1 sources, ensure the warning text includes
+        all the journalist desginators.
+        """
+        sources = set()
+        names = [
+            "source one",
+            "source two",
+            "source three",
+            "source four",
+            "source five",
+            "source six",
+            "source seven",
+        ]
+
+        for item in names:
+            source = factory.Source(journalist_designation=item)
+            sources.update([source])
+
+        dialog = DeleteSourceDialog(sources)
 
-    def test_displays_important_information_when_shown(self):
-        assert "not be able to send them replies" in self.dialog.text()
-        assert "source will not be able to log in" in self.dialog.text()
-        assert "files and messages from that source will also be destroyed" in self.dialog.text()
+        assert all(n in dialog.make_body_text(sources) for n in names), dialog.make_body_text(
```

Comment:
I would prefer if we avoid using `all` in tests because it makes it harder to debug failures ("Failed asserting True is False", etc.). Unrolling it wouldn't be too bad IMO:
```python
        text = dialog.make_body_text(sources)
        for name in names:
            assert name in text
```

Reply:
thanks, addressed in https://github.com/freedomofpress/securedrop-client/pull/2188/commits/1e541a6694678839c20305ff9084bd3f7e0f7cf6